pkg/apis/core/types.go

```diff
@@ -3658,6 +3658,13 @@ type ServiceSpec struct {
 	// If this is not specified or empty, no topology constraints will be applied.
 	// +optional
 	TopologyKeys []string
+
+	// allocateLoadBalancerNodePorts defines if NodePorts shall be
+	// automatically for services with type LoadBalancer.  Default is
+	// "true". It may be set to "false" in clusters that don't have an
```

Comment:
It may be set to "false" if the cluster load-balancer does not rely on NodePorts.